project_task_category/models/project_task_category.py

```diff
@@ -11,4 +11,5 @@ class ProjectTask(models.Model):
     categ_id = fields.Many2one(
         'project.category',
         'Category',
+        domain=[('code', '=', 'categ_id')]
```

Comment:
OK, I had a deeper look.
I seems that the point is to have only a subset of Categories available for the m2o field.
Thinking about it, in general it makes sense that the Categories from the m2o and m2m are different, since they will fit different purposes in your implementation.

We could add some type of flag on Categories to make this distinction.
But we should also add the domain to the Tags field, but that can conflict with the `project_categ` module :-/
I suggest that you "copy" `project.categ` into a new model (`project.categ.main`?), using prototype inheritance:
```
class ProjectCategMain(models.Model):
    _name = `project.categ.main`
    _inherit = `project.categ`
```

This also makes easy to have different ACLs on each Category group: main Categories can be read only for regular users, and Tags Categories can be read-write, for example, to support dynamic reusable keywords.